editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -0,0 +1,234 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+
+import { useAtom } from 'jotai'
+import React from 'react'
+import { matchRoutes } from 'react-router'
+import { uniqBy } from '../../../core/shared/array-utils'
+import * as EP from '../../../core/shared/element-path'
+import { NO_OP } from '../../../core/shared/utils'
+import { FlexColumn, FlexRow, Icn, Subdued, UtopiaTheme, colorTheme } from '../../../uuiui'
+import { RemixIndexPathLabel } from '../../canvas/remix/remix-utils'
+import {
+  ActiveRemixSceneAtom,
+  RemixNavigationAtom,
+} from '../../canvas/remix/utopia-remix-root-component'
+import { Substores, useEditorState } from '../../editor/store/store-hook'
+import { ExpandableIndicator } from '../navigator-item/expandable-indicator'
+import { getFeaturedRoutesFromPackageJSON } from '../../../printer-parsers/html/external-resources-parser'
+import { defaultEither } from '../../../core/shared/either'
+
+type RouteMatch = {
+  path: string
+  resolvedPath: string
+  matchesRealRoute: boolean
+}
+
+export const PagesPane = React.memo((props) => {
+  const featuredRoutes = useEditorState(
+    Substores.projectContents,
+    (store) => {
+      return defaultEither([], getFeaturedRoutesFromPackageJSON(store.editor.projectContents))
+    },
+    'PagesPane featuredRoutes',
+  )
+  const registeredExampleRoutes = fillInGapsInRoute(featuredRoutes)
+
+  const remixRoutes: Array<RouteMatch> = useEditorState(
+    Substores.derived,
+    (store) => {
+      const result = uniqBy(
+        registeredExampleRoutes.flatMap((exampleRoute): Array<RouteMatch> => {
+          const matchResult = matchRoutes(store.derived.remixData?.routes ?? [], exampleRoute) ?? []
+
+          if (
+            matchResult[matchResult.length - 1]?.pathname !==
+            matchResult[matchResult.length - 1]?.pathnameBase
+          ) {
+            return [
+              {
+                path: exampleRoute,
+                resolvedPath: exampleRoute,
+                matchesRealRoute: false,
+              },
+            ]
+          }
+
+          return matchResult?.map(
```

Comment:
As far as I see `matchResult` can not be null (it is initialized with `?? []`), so the `?` operator is unnecessary